lib/index.js

```diff
@@ -36,18 +36,28 @@ module.exports = function(opts) {
       ctx.throw(401, 'Invalid secret\n');
     }
 
+    let isRevoked = opts.isRevoked || (() => Promise.resolve());
+
```

Comment:
If the revocation handler is moved into a named function, and the `let isRevoked ..` line is slightly refactored, then we can prevent the performance hit of the creation of two resolved promises and the execution of the handler from happening for the 95% (completely evidence-less estimate 😀 ) of users that don't use this feature:

```
const revocationHandler = revoked => {
          if (revoked) {
            return Promise.reject(new Error('Revoked token'));
          }
          return Promise.resolve(user)
}

const isRevoked = opts.isRevoked
    ? user => opts.isRevoked(user).then(revocationHandler)
    : user => user;

return JWT.verifyAsync(token, secret, opts)
   .then(isRevoked)
   .then((user) => {
       ctx.state = ctx.state || {};
       ctx.state[opts.key] = user;
   })
   // ...
```

NB this is not tested 😀